internal/cmd/temporal-features-test-proxy/main.go

```diff
@@ -0,0 +1,752 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+)
+
+const HelpText = `The test proxy exposes the following control endpoints:
+
+- POST /quit
+  Shut down the proxy and exit.
+
+- POST /restart
+  Gracefully shut down the gRPC server, then start it again.
+
+  - Query param: sleep=<duration>
+    Forces the restart to block for the given duration; default: 0s.
+
+  - Query param: forceful=<bool>
+    If true, forces a non-graceful shutdown; default: false.
+
+- POST /reject
+  Immediately reject incoming gRPC requests with UNAVAILABLE.
+
+- POST /accept
+  Accept incoming gRPC requests; this is the default.
+
+- POST /freeze
+  Block on incoming accepted gRPC requests.
+
+- POST /thaw
+  Process incoming accepted gRPC requests immediately; this is the default.
+`
+
+var ErrUnknownCommand = errors.New("unknown command")
+
+var (
+	flagTrace   bool
```

Comment:
A bit in the same vein as asking why this is its own binary, it might be nice to move all of these into some `FeaturesProxyServer` struct or something